sqlite-db/src/insert_rollover_completed_event_data.rs

```diff
@@ -0,0 +1,416 @@
+                    .await?;
+            }
+
+            for (_event_id, cets) in dlc.cets {
+                // ignoring event_id for now because we only have one event
+                for cet in cets {
+                    insert_cet(&mut inner_transaction, event.id, cet).await?;
+                }
+            }
+
+            // Commit the transaction to either write all or rollback
+            inner_transaction.commit().await?;
+        }
+        EventKind::RolloverCompleted { dlc: None, .. } => {
+            // We ignore rollover completed events without DLC data as we don't need to store
+            // anything
+        }
+        _ => {
+            tracing::error!("Invalid event type. Use `append_event` function instead")
+        }
+    }
+
+    Ok(())
+}
+
+/// Inserts RolloverCompleted data and returns the resulting rowid
+async fn delete_rollover_completed_event_data(
+    inner_transaction: &mut Transaction<'_, Sqlite>,
+    offer_id: OrderId,
+) -> Result<()> {
+    sqlx::query(
+        r#"
+            delete from rollover_completed_event_data where cfd_id = (select id from cfds where cfds.uuid = $1)
+        "#,
+    ).bind(offer_id).execute(&mut *inner_transaction)
+        .await
+        .with_context(|| format!("Failed to delete from rollover_completed_event_data for {offer_id}"))?;
+
+    sqlx::query(
+        r#"
+            delete from revoked_commit_transactions where cfd_id = (select id from cfds where cfds.uuid = $1)
+        "#,
+    ).bind(offer_id).execute(&mut *inner_transaction)
+        .await
+        .with_context(|| format!("Failed to delete from revoked_commit_transactions for {offer_id}"))?;
+
+    sqlx::query(
+        r#"
+            delete from open_cets where cfd_id = (select id from cfds where cfds.uuid = $1)
+        "#,
+    )
+    .bind(offer_id)
+    .execute(&mut *inner_transaction)
+    .await
+    .with_context(|| format!("Failed to delete from open_cets for {offer_id}"))?;
+
+    Ok(())
+}
+
+/// Inserts RolloverCompleted data and returns the resulting rowid
+async fn insert_rollover_completed_event_data(
+    inner_transaction: &mut Transaction<'_, Sqlite>,
+    event_id: i64,
+    dlc: &Dlc,
+    funding_fee: FundingFee,
+    offer_id: OrderId,
+) -> Result<()> {
+    let (lock_tx, lock_tx_descriptor) = dlc.lock.clone();
+    let (commit_tx, commit_adaptor_signature, commit_descriptor) = dlc.commit.clone();
+    let (refund_tx, refund_signature) = dlc.refund.clone();
+
+    // casting because u64 is not implemented for sqlx: https://github.com/launchbadge/sqlx/pull/919#discussion_r557256333
+    let funding_fee_as_sat = funding_fee.fee.as_sat() as i64;
+    // TODO: these seem to be redundant and should be in `cfds` table only
+    let maker_lock_amount = dlc.maker_lock_amount.as_sat() as i64;
+    let taker_lock_amount = dlc.taker_lock_amount.as_sat() as i64;
+
+    let query_result = sqlx::query(
+        r#"
+            insert into rollover_completed_event_data (
```

Comment:
🙃 I find the name of this table weird - I think it is a result of what currently exists and does not reflect what it actually represent. Maybe at some point we can rename it to `current_contract_data` or similar. 
Does not have to be actioned as part of this PR.

Reply:
Why? It only includes `data` for the event `RolloverCompleted`, i.e. it moves `data` from `events` into this table.

Reply:
Ah right, that's what you meant in the PR description. Fair enough then!